sale_commission_so_based/models/commission_settlement.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2014-2022 Tecnativa - Pedro M. Baeza
+from odoo import api, fields, models
+
+
+class CommissionSettlement(models.Model):
+    _inherit = "commission.settlement"
+
+    settlement_type = fields.Selection(
+        selection_add=[("sale_order", "Sales Orders")],
+        ondelete={"sale_order": "set default"},
+    )
+
+
+class SettlementLine(models.Model):
+    _inherit = "commission.settlement.line"
+
+    sale_agent_line_id = fields.Many2one(comodel_name="sale.order.line.agent")
+    sale_line_id = fields.Many2one(
+        comodel_name="sale.order.line",
+        store=True,
+        related="sale_agent_line_id.object_id",
+        string="Source sale line",
+    )
+
+    @api.depends("invoice_agent_line_id", "sale_agent_line_id")
```

Comment:
```suggestion
    @api.depends("sale_agent_line_id")
```
no need to repeat the names of fields @depnds has cumulative effect

Reply:
ok